test/test_logging.py

```diff
@@ -0,0 +1,177 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Test Nature logging methods
+"""
+from typing import List, Dict, Set
+import sys
+import unittest
+import logging
+import tempfile
+import warnings
+from test import QiskitNatureTestCase
+from qiskit import BasicAer
+from qiskit.algorithms import VQE
+from qiskit.algorithms.optimizers import L_BFGS_B
+from qiskit.circuit.library import TwoLocal
+from qiskit_nature import settings as nature_settings
+from qiskit_nature.circuit.library import HartreeFock
+from qiskit_nature.drivers import UnitsType
+from qiskit_nature.drivers.second_quantization import PySCFDriver
+from qiskit_nature.problems.second_quantization.electronic import ElectronicStructureProblem
+from qiskit_nature.mappers.second_quantization import ParityMapper
+from qiskit_nature.converters.second_quantization import QubitConverter
+import qiskit_nature.optionals as _optionals
+
+
+class TestHandler(logging.StreamHandler):
+    """Unit Test Handler"""
+
+    def __init__(self):
+        """
+        Initialize the handler.
+        """
+        super().__init__(sys.stdout)
+        self.records: List[logging.LogRecord] = []
+
+    def emit(self, record) -> None:
+        """handle record"""
+        self.records.append(record)
+
+
+class TestLogging(QiskitNatureTestCase):
+    """Test logging"""
+
+    def setUp(self):
+        super().setUp()
+        self._test_handler = TestHandler()
+        nature_settings.dict_aux_operators = True
+        self._logging_dict = {"qiskit_nature": logging.DEBUG, "qiskit": logging.DEBUG}
+        self._old_logging_dict = nature_settings.logging.get_levels_for_names(
+            self._logging_dict.keys()
+        )
+
+        nature_settings.logging.set_levels_for_names(self._logging_dict, add_default_handler=False)
```

Comment:
I see the default handler being disabled here but do not see it being tested anywhere with it actually set.
That should be possible using [unittest.assertLogs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs). Maybe that even alleviates the need for `TestHandler`?